src/mechanics/constraintsPde/Equation.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <vector>
+#include "mechanics/constraintsPde/Term.h"
+
+namespace NuTo
+{
+namespace ConstraintPde
+{
+typedef std::function<double(double)> RhsFunction;
+
+//! @brief stores a constraint equation
+class Equation
+{
+public:
+    //! @brief ctor with constant rhs, defaults to 0
+    //! @param dependentNode node reference
+    //! @param dependentComponent component in the dof vector of the node
+    //! @param rhs value for the constant rhs
+    Equation(const NodeSimple& dependentNode, int dependentComponent, RhsFunction rhs)
```

Comment:
This will cause an evil bug. 
We could have a method `AreAllConstrainedTermOfDofType(DofType, mesh)` or something.

Reply:
Btw, I would not check that automatically since this would introduce a lot of dependencies. We could have:
~~~cpp
// in main file, if needed:
Group<Node> dofNodes = mesh.NodesTotal(dof);
Group<Node> constrainedDofNodes = constrains.ConstrainedNodes(dof);
if (not Group::Difference(constrainedDofNodes, dofNodes).Empty())
   throw "You messed up.";
~~~